### DISCUSSION TITLE
Downgrading to latest version compatible to python3.7

### DISCUSSION THREAD

**Discussion by renyhp** [searxng/searxng#3331] [Q&A] [answered]
Hi,

I just updated my instance, but I got a lot of weird errors after the instance update.

I then realized that I am on debian buster and my python version is 3.7 which, according to a discussion here, is not supported anymore.

I do realize I need to upgrade my OS, but, as a hotfix, do I have the possibility of downgrading to the previous version (which was working pretty well anyway), or any version that actually works with my setup?

Thank you very much in advance.

**Comment by unixfox** [comment]
3.7 is very old. I would consider upgrading instead of downgrading.

**Reply by renyhp** [reply]
Yes, as I was saying, I am strongly considering this. But as far as I understand there is no later python version available in debian buster, so I have to upgrade my OS in order to upgrade python. I surely am going to do it, but this takes time. In the meantime I would like, if possible, to have a working version of searxng rather than the currently installed broken one.

**Comment by renyhp** [comment, marked as answer]
Well ok that was easier than I expected. I did a `git checkout` of e67a7b808 which is just before the merge of the branch that dropped the python3.7 compatibility. Then I simply ran (as `searxng` user, otherwise I would get some `Permission denied`) `pip install -r requirements.txt`. searxng is working again now.

I will soon upgrade my OS and move back to the current searxng version.

Thanks anyway!
